app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java

```diff
@@ -122,6 +124,7 @@ public class SearchRepositoryImpl implements SearchRepository {
 
     private final MetadataIconProvider metadataIconProvider;
     private final ProfilePictureProvider profilePictureProvider;
+    private CustomIntentRepository customIntentRepository;
```

Comment:
I think that repositories shouldn't be dependent on other repositories, domain logic (use cases, interactors...) should handle that, anyway given that this code need to be refactored, if you consider is the best solution for now, it's OK for me 😄 .